client/lib/plugins/utils.js

```diff
@@ -340,9 +340,13 @@ export function getPluginAuthorProfileKeyword( plugin ) {
 /**
  * @param currentPlan
  * @param pluginBillingPeriod
+ * @param desiredPlan
  * @returns the correct plan slug depending on current plan and pluginBillingPeriod
  */
-export function marketplacePlanToAdd( currentPlan, pluginBillingPeriod ) {
+export function marketplacePlanToAdd( currentPlan, pluginBillingPeriod, desiredPlan ) {
+	if ( desiredPlan ) {
+		return desiredPlan;
+	}
```

Comment:
This change looks kind unnecessary, can you just used the desired plan if you have it?

Also this is blocking: Adding a new param to this function isn't safe until this dead code gets cleaned up: https://github.com/Automattic/wp-calypso/pull/68178

Reply:
Sounds good, [I fixed it here](https://github.com/Automattic/wp-calypso/pull/68154/commits/709f78d0e2825d8680ea4bb310d13767fa95a1eb).